### PR TITLE
Investigate whether Google Apps intermediary page can be avoided

### PR DESCRIPTION
**Pull request by sallakarppinen**
Currently, clicking the "Sign In" button on a tenant that's configured with Google Apps authentication causes you to be sent off to the general Google Auth page.

screen shot 2015-12-28 at 18 10 56

The user is then expected to enter an email address for that Google Apps domain, without any indication of them having to do so. When signing in with a different Google address, no error message is provided either.

We should investigate if we can:

- Send them directly to the SSO page Google forwards them to (preferred)
- Provide indication of what address they should use

### CODE REVIEW

File: node_modules/oae-authentication/lib/strategies/google/init.js [hunk 1]
```diff
@@ -62,7 +63,14 @@ module.exports = function() {
                 'email'
             ],
             'callbackURL': AuthenticationUtil.constructCallbackUrl(tenant, AuthenticationConstants.providers.GOOGLE)
-        },  function(req, accessToken, refreshToken, profile, done) {
+        };
+
+        // If there's only one allowed domain, add that to options as the hosted domain
+        if(domains.length === 1) {
```
Comment (nicolaasmatthijs) [inline node_modules/oae-authentication/lib/strategies/google/init.js]: Need a space after `if` to comply with styleguide

File: node_modules/oae-authentication/lib/strategies/google/init.js [hunk 2]
```diff
@@ -62,7 +63,14 @@ module.exports = function() {
                 'email'
             ],
             'callbackURL': AuthenticationUtil.constructCallbackUrl(tenant, AuthenticationConstants.providers.GOOGLE)
-        },  function(req, accessToken, refreshToken, profile, done) {
+        };
+
+        // If there's only one allowed domain, add that to options as the hosted domain
```
Comment (nicolaasmatthijs) [inline node_modules/oae-authentication/lib/strategies/google/init.js]: Can you add an `@see` to the documentation of the `hd` parameter here?

**Comment by nicolaasmatthijs** [comment]
Left some minor comments. Back to @sallakarppinen for follow-up

**Comment by sallakarppinen** [comment]
Fixed the style issue and added a link to comment.

**Comment by nicolaasmatthijs** [comment]
Looks good. Merged.

**Pull request merged**